xbmc/video/VideoDatabase.cpp

```diff
@@ -6184,11 +6185,55 @@ void CVideoDatabase::UpdateTables(int iVersion)
         "INSERT INTO videoversion SELECT idFile, idMovie, 'movie', '%i', '%i' FROM movie",
         VideoAssetType::VERSION, VIDEO_VERSION_ID_DEFAULT));
   }
+
+  if (iVersion < 127)
+  {
+    m_pDS->exec("ALTER TABLE videoversiontype ADD itemType INTEGER");
+
+    // First, assume all types are video version types
+    m_pDS->exec(PrepareSQL("UPDATE videoversiontype SET itemType = %i", VideoAssetType::VERSION));
+
+    // Then, check current extras entries and their assigned item type and migrate it
+
+    // get all assets with extras item type
+    m_pDS->query("SELECT DISTINCT idType FROM videoversion WHERE itemType = 1");
+    while (!m_pDS->eof())
+    {
+      const int idType{m_pDS->fv(0).get_asInt()};
+      if (idType > VIDEO_VERSION_ID_END)
+      {
+        // user-added type for extras. change its item type to extras
+        m_pDS2->exec(PrepareSQL("UPDATE videoversiontype SET itemType = %i WHERE id = %i",
+                                VideoAssetType::EXTRAS, idType));
+      }
+      else
+      {
+        // system type used for an extra. copy as extras item type.
```

Comment:
I understand what you're going for, but I don't think it's possible for the situation to happen?
The code prior to this PR (up to db v126) precreated all records from id 40400 to 40800, reserving them for versions (therefore excluding extras from that range).
Unless I missed a place in code where extras management code reuses an id between 40400 to 40800 instead of allocating a new id.

Reply:
Rename an extra, pick a system provided version, like „standard edition“ and there you go. This is very well possible, unfortunately. I tried that out.

Reply:
> precreated all records from id 40400 to 40800, reserving them for versions (therefore excluding extras from that range).

The last part of your statement is wrong. Before this PR, the types were agnostic to versions / extras. That’s the actual problem the PR fixed. The reservation was not for versions, only for „system provided“.